realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -1564,4 +1583,121 @@ public void setter_nullValueInNullableField() {
         realm.commitTransaction();
         assertNull(realm.allObjects(NullTypes.class).first().getFieldDateNull());
     }
+
+    @Test
+    @RunTestInLooperThread
+    public void addChangeListener_throwOnAddingNullListener() {
+        final Realm realm = looperThread.realm;
+        realm.beginTransaction();
+        Dog dog = realm.createObject(Dog.class);
+        dog.setAge(2);
+        realm.commitTransaction();
+
+        try {
+            dog.addChangeListener((RealmChangeListener) null);
+            fail("Failed on adding null change listener.");
+        } catch (IllegalArgumentException ignore) {
+            looperThread.testComplete();
+        }
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void addChangeListener_throwOnNullRealm() {
+        final Realm realm = looperThread.realm;
+        realm.beginTransaction();
+        Dog dog = new Dog();
+        dog.setAge(2);
+        realm.commitTransaction();
+
+        try {
+            dog.addChangeListener(new RealmChangeListener() {
+                @Override
+                public void onChange() {
+                }
+            });
+            fail("Failed on adding listener on null realm.");
+        } catch (IllegalArgumentException ignore) {
+            looperThread.testComplete();
+        }
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void removeChangeListener_throwOnRemovingNullListener() {
+        final Realm realm = looperThread.realm;
+        realm.beginTransaction();
+        Dog dog = realm.createObject(Dog.class);
+        dog.setAge(2);
+        realm.commitTransaction();
+        RealmChangeListener listener = new RealmChangeListener() {
+            @Override
+            public void onChange() {
+            }
+        };
+        dog.addChangeListener(listener);
+
+        try {
+            dog.removeChangeListener((RealmChangeListener) null);
+            fail("Failed on adding null change listener.");
+        } catch (IllegalArgumentException ignore) {
+            looperThread.testComplete();
+        }
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void removeChangeListeners() {
```

Comment:
I'd love to avoid mundane tests, but unfortunately `removeChangeListeners()` isn't tested on  `RealmObject`. (I've searched all the tests and I haven't found one.) I'll remove this test if it still doesn't make sense.